website/blog/2024-03-18-deploying-offline-with-docker/index.md

```diff
@@ -0,0 +1,52 @@
+---
+slug: deploying-offline-with-docker
+title: "Deploying offline with Docker"
+authors: [wwayne]
+tags: [deployment offline]
+---
+
+<div align="center">
+
+![No internet access](./no-internet.png)
+
+</div>
+
+Are you facing a situation where your work environment lacks internet access, and you're wondering if you can still deploy Tabby? Fear not, because the answer is YES! 🐱📣
+
+## Prerequisite📋
+* Docker installed on both the internet-connected computer and the offline computer.
+
+## The Game Plan🚀
+* Create a Docker image on a computer with internet access. 
+* Transfer the image to your offline computer.
+* Run the Docker image and let Tabby work its magic! ✨
+
+## Step-by-Step Guide🐾
+1. Create a new **Dockerfile** on a computer with internet access.
+```docker
+FROM tabbyml/tabby
+
+RUN /opt/tabby/bin/tabby-cpu download --model TabbyML/StarCoder-1B
```

Comment:
use environment [TABBY_MODEL_CACHE_ROOT](https://github.com/TabbyML/tabby/blob/52badbbc50b26d897caccda6efc050c9cacd2325/crates/tabby-common/src/path.rs#L12) to store models in `/models` directory

So user can still mount `/data` directory for logging / events if needed

Reply:
Added `ENV TABBY_MODEL_CACHE_ROOT=/models` to the Dockerfile